<a href="https://colab.research.google.com/github/DTalis/py/blob/main/Exercise2_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd


In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Time_Americans_Spend_Sleeping.csv')
df

index  Year  Period  Avg hrs per day sleeping  Standard Error  \
0        0  2003  Annual                      8.57           0.018   
1        1  2004  Annual                      8.55           0.026   
2        2  2005  Annual                      8.62           0.023   
3        3  2006  Annual                      8.63           0.024   
4        4  2007  Annual                      8.57           0.024   
..     ...   ...     ...                       ...             ...   
940    940  2013  Annual                      8.83           0.085   
941    941  2014  Annual                      8.98           0.086   
942    942  2015  Annual                      8.83           0.074   
943    943  2016  Annual                      8.92           0.071   
944    944  2017  Annual                      8.81           0.091   

            Type of Days          Age Group  Activity    Sex  
0               All days  15 years and over  Sleeping   Both  
1               All days  15 years and over  Sleeping   Both  
2               All days  15 years and over  Sleeping   Both  
3               All days  15 years and over  Sleeping   Both  
4               All days  15 years and over  Sleeping   Both  
..                   ...                ...       ...    ...  
940  Nonholiday weekdays  65 years and over  Sleeping  Women  
941  Nonholiday weekdays  65 years and over  Sleeping  Women  
942  Nonholiday weekdays  65 years and over  Sleeping  Women  
943  Nonholiday weekdays  65 years and over  Sleeping  Women  
944  Nonholiday weekdays  65 years and over  Sleeping  Women  

[945 rows x 9 columns]

In [ ]:
df.head()

index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both

The provided dataset, Time Americans Spend Sleeping.csv, contains information on the average number of hours Americans spent sleeping from 2003 to 2017.

The dataset has 945 entries and 9 columns, including:

Year: The year the data was recorded.

Period: The time period (e.g., Annual).

Avg hrs per day sleeping: The average hours spent sleeping per day.

Standard Error: The standard error of the mean.

Type of Days: Whether the data is for All days, Nonholiday weekdays, or Weekend days and holidays.

Age Group: The age group of the participants (e.g., 15 years and over, 25 to 34 years).

Activity: The activity being measured (in this case, always 'Sleeping').

Sex: The gender of the participants (Both, Men, or Women).


Quantitative and Qualitative Column Breakdown:

Quantitative Columns: index, Year, Avg hrs per day sleeping, Standard Error

Qualitative Columns: Period, Type of Days, Age Group, Activity, Sex

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Mental_health_Depression_disorder_Data.csv')
df

/tmp/ipython-input-248328894.py:1: DtypeWarning: Columns (5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Mental_health_Depression_disorder_Data.csv')


index       Entity Code  Year Schizophrenia (%) Bipolar disorder (%)  \
0            0  Afghanistan  AFG  1990           0.16056             0.697779   
1            1  Afghanistan  AFG  1991          0.160312             0.697961   
2            2  Afghanistan  AFG  1992          0.160135             0.698107   
3            3  Afghanistan  AFG  1993          0.160037             0.698257   
4            4  Afghanistan  AFG  1994          0.160022             0.698469   
...        ...          ...  ...   ...               ...                  ...   
108548  108548     Zimbabwe  ZWE  2013      303564.60359                  NaN   
108549  108549     Zimbabwe  ZWE  2014     311665.769283                  NaN   
108550  108550     Zimbabwe  ZWE  2015     320638.507158                  NaN   
108551  108551     Zimbabwe  ZWE  2016     330437.353798                  NaN   
108552  108552     Zimbabwe  ZWE  2017     340583.157405                  NaN   

       Eating disorders (%)  Anxiety disorders (%)  Drug use disorders (%)  \
0                  0.101855               4.828830                1.677082   
1                  0.099313               4.829740                1.684746   
2                  0.096692               4.831108                1.694334   
3                  0.094336               4.830864                1.705320   
4                  0.092439               4.829423                1.716069   
...                     ...                    ...                     ...   
108548                  NaN                    NaN                     NaN   
108549                  NaN                    NaN                     NaN   
108550                  NaN                    NaN                     NaN   
108551                  NaN                    NaN                     NaN   
108552                  NaN                    NaN                     NaN   

        Depression (%)  Alcohol use disorders (%)  
0             4.071831                   0.672404  
1             4.079531                   0.671768  
2             4.088358                   0.670644  
3             4.096190                   0.669738  
4             4.099582                   0.669260  
...                ...                        ...  
108548             NaN                        NaN  
108549             NaN                        NaN  
108550             NaN                        NaN  
108551             NaN                        NaN  
108552             NaN                        NaN  

[108553 rows x 11 columns]

In [ ]:
df.head()

index       Entity Code  Year Schizophrenia (%) Bipolar disorder (%)  \
0      0  Afghanistan  AFG  1990           0.16056             0.697779   
1      1  Afghanistan  AFG  1991          0.160312             0.697961   
2      2  Afghanistan  AFG  1992          0.160135             0.698107   
3      3  Afghanistan  AFG  1993          0.160037             0.698257   
4      4  Afghanistan  AFG  1994          0.160022             0.698469   

  Eating disorders (%)  Anxiety disorders (%)  Drug use disorders (%)  \
0             0.101855               4.828830                1.677082   
1             0.099313               4.829740                1.684746   
2             0.096692               4.831108                1.694334   
3             0.094336               4.830864                1.705320   
4             0.092439               4.829423                1.716069   

   Depression (%)  Alcohol use disorders (%)  
0        4.071831                   0.672404  
1        4.079531                   0.671768  
2        4.088358                   0.670644  
3        4.096190                   0.669738  
4        4.099582                   0.669260

This dataset provides information on the prevalence of several mental health disorders across various countries and years.

The dataset contains 108,553 rows and 11 columns. The columns are:

index: A unique identifier for each row.

Entity: The name of the country or region.

Code: The three-letter country code.

Year: The year of the data entry.

Schizophrenia (%): The percentage of the population with schizophrenia.

Bipolar disorder (%): The percentage of the population with bipolar disorder.

Eating disorders (%): The percentage of the population with eating disorders.

Anxiety disorders (%): The percentage of the population with anxiety disorders.

Drug use disorders (%): The percentage of the population with drug use disorders.

Depression (%): The percentage of the population with depression.

Alcohol use disorders (%): The percentage of the population with alcohol use disorders.

It is important to note that many of the columns have missing values and some of the percentage columns (Schizophrenia (%), Bipolar disorder (%), Eating disorders (%)) are of the object data type and may contain non-numeric characters that need to be cleaned before they can be used for numerical analysis.

Quantitative Columns:
index:
Year:
Anxiety disorders (%)
Drug use disorders (%)
Depression (%)
Alcohol use disorders (%)
Schizophrenia (%)
Bipolar disorder (%)
Eating disorders (%)

Qualitative Columns: Entity, Code.

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/.ipynb_checkpoints/clean_dataset.csv')
df

Gender    Age    Debt  Married  BankCustomer         Industry Ethnicity  \
0         1  30.83   0.000        1             1      Industrials     White   
1         0  58.67   4.460        1             1        Materials     Black   
2         0  24.50   0.500        1             1        Materials     Black   
3         1  27.83   1.540        1             1      Industrials     White   
4         1  20.17   5.625        1             1      Industrials     White   
..      ...    ...     ...      ...           ...              ...       ...   
685       1  21.08  10.085        0             0        Education     Black   
686       0  22.67   0.750        1             1           Energy     White   
687       0  25.25  13.500        0             0       Healthcare    Latino   
688       1  17.92   0.205        1             1  ConsumerStaples     White   
689       1  35.00   3.375        1             1           Energy     Black   

     YearsEmployed  PriorDefault  Employed  CreditScore  DriversLicense  \
0             1.25             1         1            1               0   
1             3.04             1         1            6               0   
2             1.50             1         0            0               0   
3             3.75             1         1            5               1   
4             1.71             1         0            0               0   
..             ...           ...       ...          ...             ...   
685           1.25             0         0            0               0   
686           2.00             0         1            2               1   
687           2.00             0         1            1               1   
688           0.04             0         0            0               0   
689           8.29             0         0            0               1   

          Citizen  ZipCode  Income  Approved  
0         ByBirth      202       0         1  
1         ByBirth       43     560         1  
2         ByBirth      280     824         1  
3         ByBirth      100       3         1  
4    ByOtherMeans      120       0         1  
..            ...      ...     ...       ...  
685       ByBirth      260       0         0  
686       ByBirth      200     394         0  
687       ByBirth      200       1         0  
688       ByBirth      280     750         0  
689       ByBirth        0       0         0  

[690 rows x 16 columns]

In [ ]:
df.head()

Gender    Age   Debt  Married  BankCustomer     Industry Ethnicity  \
0       1  30.83  0.000        1             1  Industrials     White   
1       0  58.67  4.460        1             1    Materials     Black   
2       0  24.50  0.500        1             1    Materials     Black   
3       1  27.83  1.540        1             1  Industrials     White   
4       1  20.17  5.625        1             1  Industrials     White   

   YearsEmployed  PriorDefault  Employed  CreditScore  DriversLicense  \
0           1.25             1         1            1               0   
1           3.04             1         1            6               0   
2           1.50             1         0            0               0   
3           3.75             1         1            5               1   
4           1.71             1         0            0               0   

        Citizen  ZipCode  Income  Approved  
0       ByBirth      202       0         1  
1       ByBirth       43     560         1  
2       ByBirth      280     824         1  
3       ByBirth      100       3         1  
4  ByOtherMeans      120       0         1

This dataset appears to be related to credit card applications, containing various demographic and financial details about the applicants.

The dataset has 690 rows and 16 columns, with no missing values. The columns include:

Gender: Applicant's gender (represented as 0 or 1).

Age: Applicant's age.

Debt: Applicant's total debt.

Married: Marital status (represented as 0 or 1).

BankCustomer: Indicates if the applicant is a bank customer (represented as 0 or 1).

Industry: Applicant's industry of employment.

Ethnicity: Applicant's ethnicity.

YearsEmployed: Number of years the applicant has been employed.

PriorDefault: Indicates if the applicant has a prior credit default (represented as 0 or 1).

Employed: Employment status (represented as 0 or 1).

CreditScore: Applicant's credit score.

DriversLicense: Indicates if the applicant has a driver's license (represented as 0 or 1).

Citizen: Applicant's citizenship status.

ZipCode: Applicant's zip code.

Income: Applicant's income.

Approved: The target variable indicating whether the application was approved (1) or not (0).

Quantitative Columns:
Age,Debt, YearsEmployed, CreditScore, Income, ZipCode: This column contains numerical values that are used for identification, even though they are a form of code, they represent a specific location that can be ordered in some cases.

Qualitative Columns:
Gender, Married, BankCustomer, PriorDefault, Employed, DriversLicense, Approved, Industry, Ethnicity, Citizen